### ESTIMATION AND CONFIDENCE INTERVALS

#### Data 
A total of 15 print-heads were randomly selected and tested until failure. The durability of each print-head (in millions of characters) was recorded as follows: 

1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48, 1.20, 1.33, 1.18, 1.22, 1.29

#### Assignment Tasks 
##### 1)  Build 99% Confidence Interval Using Sample Standard Deviation 

Assuming the sample is representative of the population, construct a 99% confidence interval for the mean number of characters printed before the print-head fails using the sample standard deviation. Explain the steps you take and the rationale behind using the t-distribution for this task.

##### 2) Build 99% Confidence Interval Using Known Population Standard Deviation

If it were known that the population standard deviation is 0.2 million characters, construct a 99% confidence interval for the mean number of characters printed before failure.

We will use:

* t-distribution when population standard deviation is unknown (sample standard deviation used).

* z-distribution when population standard deviation is known.

In [14]:
# Confidence Intervals for Print-Head Durability
import numpy as np
import scipy.stats as stats

In [15]:
# Sample data (durability in millions of characters)
data = np.array([1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 
                 0.85, 1.07, 1.48, 1.20, 1.33, 1.18, 1.22, 1.29])

In [16]:
# Basic statistics
n = len(data)                   # sample size
mean = np.mean(data)            # sample mean
s = np.std(data, ddof=1)        # sample standard deviation (ddof=1 for unbiased)

In [17]:
print("Sample Size (n):", n)
print("Sample Mean (millions):", round(mean, 4))
print("Sample Std Dev (s, millions):", round(s, 4))

Sample Size (n): 15
Sample Mean (millions): 1.2387
Sample Std Dev (s, millions): 0.1932


In [18]:
# 99% confidence
alpha = 0.01

In [19]:
# (a) Using t-distribution (sigma unknown)--> using sample standard deviation
t_critical = stats.t.ppf(1 - alpha/2, df=n-1)     # two-tailed critical t
margin_error_t = t_critical * (s / np.sqrt(n))

ci_lower_t = mean - margin_error_t
ci_upper_t = mean + margin_error_t

print("\n(a) 99% CI (t-distribution):")
print("  t critical (df={}): {:.4f}".format(n-1, t_critical))
print("  Margin of error:", round(margin_error_t, 4))
print("  99% CI: ({:.4f}, {:.4f}) million characters".format(ci_lower_t, ci_upper_t))


(a) 99% CI (t-distribution):
  t critical (df=14): 2.9768
  Margin of error: 0.1485
  99% CI: (1.0902, 1.3871) million characters


In [20]:
# (b) Using z-distribution (sigma known)--> using population deviation
sigma = 0.2    # known population std dev
z_critical = stats.norm.ppf(1 - alpha/2)
margin_error_z = z_critical * (sigma / np.sqrt(n))

ci_lower_z = mean - margin_error_z
ci_upper_z = mean + margin_error_z

print("\n(b) 99% CI (z-distribution, sigma=0.2):")
print("  z critical:", round(z_critical, 4))
print("  Margin of error:", round(margin_error_z, 4))
print("  99% CI: ({:.4f}, {:.4f}) million characters".format(ci_lower_z, ci_upper_z))



(b) 99% CI (z-distribution, sigma=0.2):
  z critical: 2.5758
  Margin of error: 0.133
  99% CI: (1.1057, 1.3717) million characters


##### Results (numeric)

Sample size 𝑛 = 15
Sample mean 𝑥ˉ= 1.2387 million characters

Sample standard deviation 𝑠 = 0.1932 million characters

##### (a) 99% CI using sample standard deviation (t-distribution, df = 14):

(1.0902,1.3871) million characters

##### (b) 99% CI using known population standard deviation 𝜎= 0.2(z-distribution):

(1.1057,1.3717) million characters

##### Short interpretation

* The CI in (a) is slightly wider because we estimated the spread from the sample (using the t-distribution).

* When σ is assumed known (b), the interval is a bit narrower because the z-based margin uses the known population spread.

                                                                                      Submitted by: Meghana C Varghese